packages/excalidraw/data/restore.ts

```diff
@@ -596,7 +563,51 @@ export const restoreElements = (
     }
   }
 
-  return restoredElements;
+  // NOTE (mtolmacs): Temporary fix for extremely large arrows
+  // Need to iterate again so we have attached text nodes in elementsMap
+  return restoredElements.map((element) => {
+    if (
+      isElbowArrow(element) &&
+      element.startBinding &&
+      element.endBinding &&
+      element.startBinding.elementId === element.endBinding.elementId &&
```

Comment:
we could repair also non self-bound cases if they happen for whatever reason, no?

Reply:
Unfortunately no, as the routing breaks if the elements are super far away.